### PR TITLE
Feature/pause fix

### PR DESCRIPTION
**Pull request by silfaber**
created gamestate and its manager to manage pausing the game

### CODE REVIEW

File: Assets/Scripts/Managers/GameStateManager.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class GameStateManager : MonoBehaviour
+{
+    public static GameStateManager Instance;
```
Comment (JdeHaan2001) [inline Assets/Scripts/Managers/GameStateManager.cs]: Can you make this read-only? So something like {get; private set;}

**Review by JdeHaan2001 — CHANGES_REQUESTED** [review]
Everything looks okay, except for 1 thing that needs to be changed. If you make a singleton, it needs to be read-only so you can make it a read-only property. See comment for a suggestion on how to do it.

File: Assets/Scripts/Managers/GameStateManager.cs (hunk 1 above)
Comment (silfaber) [reply Assets/Scripts/Managers/GameStateManager.cs]: ah right, got it

**Comment by silfaber** [comment]
job done

**Review by JdeHaan2001 — APPROVED** [review]
Looks good

**Pull request merged**